epsagon/utils.py

```diff
@@ -445,3 +453,74 @@ def database_connection_type(hostname, default_type):
     if 'redshift.amazonaws' in hostname:
         return 'redshift'
     return default_type
+
+
+def obfuscate_sql_query(query, operation):
+    """
+    Obfuscate SQL queries to protect sensitive uploads
+    :param query: The SQL query string
+    :param operation: Operation in DB
+    :return: the obfuscated query (string)
+    :var bounds: specifier for locating values based upon operation
+    :var bounds: start is leftmost bound, stop is rightmost bound
+    :var bounds: separator is the symbol between values
+    :var bounds: signal is a preceding identifier for each value
+    """
+    bounds = {
```

Comment:
this parsing indeed works for some statements, but will fail on more complex scenarios - for example "select * from bla where x=1 ORDER..."  or for example "select * from bla where EXISTS (.... another statement..)
You can either: 
- catch those cases so you won't fail on obfuscating the string (I don't recommend that as it can get very complicated)
- I would suggest to look at sqlparse - it seems it will help parsing the sql query: (I didn't take a deep look though)
https://sqlparse.readthedocs.io/en/latest/analyzing/

Reply:
I looked into every sql parsing library, none of them are good enough for what we want (which is still pretty simple).
I think i'll go the first route, and only obfuscate when 100% safe

Reply:
I ended up using a bit of sqlparse & also ignoring cases we don't want to obfuscate. Looks much stronger now, and can handle very complex clauses.